File: services/academic-service/src/main/java/com/visor/school/academicservice/model/EmergencyContact.java
package com.visor.school.academicservice.model;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Emergency contact value object
 */
@Embeddable
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EmergencyContact {
    private String name;
    private String relationship;
    private String phoneNumber;
    private String email;
    private String address;
}

File: services/academic-service/src/main/kotlin/com/visor/school/academicservice/model/EmergencyContact.kt
package com.visor.school.academicservice.model

import jakarta.persistence.Embeddable

/**
 * Emergency contact value object
 */
@Embeddable
data class EmergencyContact(
    val name: String,
    val relationship: String,
    val phoneNumber: String,
    val email: String? = null,
    val address: String? = null
)

